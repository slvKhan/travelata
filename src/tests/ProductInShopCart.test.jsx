/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import ProductShopCart from '../ShopCart/ProductInShopCart';

test('ProductShopCart render_1', () => {
  const productName = 'Телефон';
  const id = 1;
  const idProduct = 4;
  const amountPrice = 10;
  const amount = 1;

  const handleremoveProduct = () => null;
  const handleincreaseProduct = () => null;
  const handlereduceProduct = () => null;
  const increaseDisabled = false;

  const product = {
    id,
    idProduct,
    productName,
    amountPrice,
    amount,
  };

  const component = renderer.create(<ProductShopCart
    key={product.idProduct}
    product={product}
    handleincreaseProduct={handleincreaseProduct}
    handlereduceProduct={handlereduceProduct}
    handleremoveProduct={handleremoveProduct}
    increaseDisabled={increaseDisabled}
  />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('ProductShopCart render_2', () => {
  const productName = 'text 2';
  const id = 51;
  const price = 2;
  const amount = 6;

  const handleremoveProduct = () => null;
  const handleincreaseProduct = () => null;
  const handlereduceProduct = () => null;
  const increaseDisabled = true;

  const product = {
    id,
    productName,
    price,
    amount,
  };

  const component = renderer.create(<ProductShopCart
    key={product.idProduct}
    product={product}
    handleincreaseProduct={handleincreaseProduct}
    handlereduceProduct={handlereduceProduct}
    handleremoveProduct={handleremoveProduct}
    increaseDisabled={increaseDisabled}
  />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
