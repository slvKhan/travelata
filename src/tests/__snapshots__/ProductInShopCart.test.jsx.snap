// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductShopCart render_1 1`] = `
<div
  className="mb-3 card"
  id={4}
>
  <div
    className="card-body"
  >
    <div
      className="content-wrap row"
    >
      <span
        className="amount pr-0 col-1"
      >
        <span>
          №
        </span>
        <span>
          1
        </span>
      </span>
      <span
        className="product-name col-3"
      >
        Телефон
      </span>
      <div
        className="quantity-controll col-3"
      >
        <button
          className="col-3 ml-1 p-0 btn-sm btn btn-outline-secondary"
          disabled={false}
          onClick={null}
          type="button"
        >
          -
        </button>
        <span>
           1 
        </span>
        <span>
          шт
        </span>
        <button
          className="col-3 ml-1 p-0 btn-sm btn btn-outline-success"
          disabled={false}
          onClick={null}
          type="button"
        >
          +
        </button>
      </div>
      <div
        className="amount-price p-0 "
      >
        <span>
          на сумму
        </span>
        <span>
           10 
        </span>
        <span>
          рублей
        </span>
      </div>
    </div>
    <div
      className="remove-wrap text-center mt-3"
    >
      <button
        className="btn-sm w-50 btn btn-outline-danger"
        disabled={false}
        onClick={null}
        type="button"
      >
        удалить
      </button>
    </div>
  </div>
</div>
`;

exports[`ProductShopCart render_2 1`] = `
<div
  className="mb-3 card"
>
  <div
    className="card-body"
  >
    <div
      className="content-wrap row"
    >
      <span
        className="amount pr-0 col-1"
      >
        <span>
          №
        </span>
        <span>
          51
        </span>
      </span>
      <span
        className="product-name col-3"
      >
        text 2
      </span>
      <div
        className="quantity-controll col-3"
      >
        <button
          className="col-3 ml-1 p-0 btn-sm btn btn-outline-secondary"
          disabled={false}
          onClick={null}
          type="button"
        >
          -
        </button>
        <span>
           6 
        </span>
        <span>
          шт
        </span>
        <button
          className="col-3 ml-1 p-0 btn-sm btn btn-outline-success"
          disabled={true}
          onClick={null}
          type="button"
        >
          +
        </button>
      </div>
      <div
        className="amount-price p-0 "
      >
        <span>
          на сумму
        </span>
        <span>
           undefined 
        </span>
        <span>
          рублей
        </span>
      </div>
    </div>
    <div
      className="remove-wrap text-center mt-3"
    >
      <button
        className="btn-sm w-50 btn btn-outline-danger"
        disabled={false}
        onClick={null}
        type="button"
      >
        удалить
      </button>
    </div>
  </div>
</div>
`;
